### DISCUSSION TITLE
Under custom script auto-sync does not sync my script

### DISCUSSION THREAD

**Discussion by dionrowney** [netbox-community/netbox#17485] [General]
I add a custom script in netbox 4.0.9 from a gitlab repository and check the autosync.

I update my script in the repo and instead of updating the script it shows me a warning saying: Data is out of sync with upstream file ([scripts/myscript.py])) and makes me press a sync button.  

Am I somehow misunderstanding "Enable automatic synchronization of data when the data file is updated"?  why is use intervention required?

**Comment by alehaa** [comment]
Yep, that's a pitfail I felt into as well. Synchronization doesn't mean it's synchronized from disk, but from the data source file on updates. You'll have to manually trigger a sync of the data source to see any updates in your script.

**Comment by Almeus** [comment]
I'm also running into this problem. I have to trigger the sync under data sources every time I update the script. 

In older versions of Netbox (c 2022) you just put scripts into the scripts folder and they showed up in the scripts section of the GUI. Now you have toadd a data source in order for them to show up.
